jms/src/main/scala/akka/stream/alpakka/jms/JmsConsumerStage.scala

```diff
@@ -254,21 +254,14 @@ abstract class SourceStageLogic[T](shape: SourceShape[T],
   })
 
   private def stopSessions(): Unit =
-    if (stopping.compareAndSet(false, true))
-      Future {
-        try {
-          jmsConnection.foreach(_.stop())
-        } catch {
-          case NonFatal(e) => log.error(e, "Error stopping JMS connection {}", jmsConnection)
-        }
-      }.flatMap { _ =>
-          val closeSessionFutures = jmsSessions.map { s =>
-            val f = s.closeSessionAsync()
-            f.onFailure { case e => log.error(e, "Error closing jms session") }
-            f
-          }
-          Future.sequence(closeSessionFutures)
-        }
+    if (stopping.compareAndSet(false, true)) {
+      val closeSessionFutures = jmsSessions.map { s =>
+        val f = s.closeSessionAsync()
+        f.failed.foreach(e => log.error(e, "Error closing jms session"))
+        f
+      }
+      Future
+        .sequence(closeSessionFutures)
         .onComplete { _ =>
```

Comment:
This is good. I covers both the success and failure cases and makes sure we close the connection in all cases.